### PR TITLE
Ch8507/application request tags front

### PR DESCRIPTION
**Pull request by ktherage**

### CODE REVIEW

File: behat.yml.dist [hunk 1]
```diff
@@ -23,6 +23,7 @@ default:
                 - SecurityContext
                 - FeatureContext
                 - DonationContext
+                - Behatch\Context\TableContext
```
Comment (thomasbisignani) [inline behat.yml.dist]: Can you declare this context after the `- Behatch\Context\XmlContext` ?

File: features/municipal_chief_space.feature [hunk 2]
```diff
@@ -1,3 +1,4 @@
+@wip
```
Comment (thomasbisignani) [inline features/municipal_chief_space.feature]: oups :man_cartwheeling:

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 3]
```diff
@@ -110,4 +116,77 @@ public function municipalVolunteerDetailAction(
...
+
+    private function handleApplicationRequestTagsType(
+        ObjectManager $objectManager,
+        Request $request,
+        ApplicationRequest $applicationRequest,
+        string $view,
+        string $redirectRoute
+    ): Response {
+        $form = $this->createForm(ApplicationRequestTagsType::class, $applicationRequest);
+        $form->handleRequest($request);
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/MunicipalChiefController.php]: Not a big deal, but you can write : 

```php
$form = $this
    ->createForm(ApplicationRequestTagsType::class, $applicationRequest)
    ->handleRequest($request)
;
```

File: templates/municipal_chief/municipal/_edit_tags.html.twig [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{% extends 'municipal_chief/_layout.html.twig' %}
+
+{% block stylesheets %}
+    <link href="{{ asset('select2/select2.min.css') }}" rel="stylesheet" />
+{% endblock %}
+
+{% block municipal_chief_content %}
+
+    {% block links %}
```
Comment (thomasbisignani) [inline templates/municipal_chief/municipal/_edit_tags.html.twig]: You can inline this block declaration : 

` {% block links %}{% endblock %}`

File: templates/referent/municipal/_edit_tags.html.twig [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{% extends 'referent/_layout.html.twig' %}
+
+{% block stylesheets %}
+    <link href="{{ asset('select2/select2.min.css') }}" rel="stylesheet" />
+{% endblock %}
+
+{% block referent_content %}
+
+    {% block links %}
```
Comment (thomasbisignani) [inline templates/referent/municipal/_edit_tags.html.twig]: Same here, you can  inline this block

File: features/referent_space.feature [hunk 6]
```diff
@@ -53,8 +53,8 @@ Feature:
     And I should see "Oui"
 
     When I click the "application-detail-0" element
-    Then I should see "Retour à la liste des candidats colistiers"
-    And I should see "Profession : Scientist"
+    And I wait 10 seconds until I see "Retour à la liste des candidats colistiers"
```
Comment (Remg) [inline features/referent_space.feature]: isn't 10 seconds a bit much ?

File: features/referent_space.feature [hunk 7]
```diff
@@ -64,6 +64,15 @@ Feature:
     And I should see "Détails du projet :"
     And I should see "Actifs professionnels :"
 
+    When I click the ".back-to-list" selector
+    And I click the "application-edit-0" element
+    Then I wait 10 seconds until I see "Tags de candidature"
```
Comment (Remg) [inline features/referent_space.feature]: same here ?

i mean these tests alone would add 1minute just waiting :p

File: src/DataFixtures/ORM/LoadApplicationRequestRunningMateRequestData.php [hunk 8]
```diff
@@ -57,8 +57,8 @@ public function load(ObjectManager $manager)
 
             $runningMateRequest->addFavoriteTheme($this->getReference('application-theme-06'));
             $runningMateRequest->addFavoriteTheme($this->getReference('application-theme-08'));
-            $runningMateRequest->addTag($this->getReference('application-request-tag-'.rand(1, 3)));
-            $runningMateRequest->addTag($this->getReference('application-request-tag-'.rand(2, 4)));
+            $runningMateRequest->addTag($this->getReference('application-request-tag-'.$i));
+            $runningMateRequest->addTag($this->getReference('application-request-tag-'.($i + 1) % 4));
```
Comment (Remg) [inline src/DataFixtures/ORM/LoadApplicationRequestRunningMateRequestData.php]: nice

File: features/referent_space.feature (hunk 6 above)
Comment (ktherage) [reply features/referent_space.feature]: it's a timeout.

File: src/DataFixtures/ORM/LoadApplicationRequestRunningMateRequestData.php (hunk 8 above)
Comment (ktherage) [reply src/DataFixtures/ORM/LoadApplicationRequestRunningMateRequestData.php]: Ah dacow

File: features/referent_space.feature (hunk 6 above)
Comment (Remg) [reply features/referent_space.feature]: :man_facepalming:  you're right my bad !

File: src/Form/ApplicationRequest/ApplicationRequestTagsType.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+class ApplicationRequestTagsType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('tags', EntityType::class, [
+                'label' => 'Tag de candidature',
+                'class' => ApplicationRequestTag::class,
+                'query_builder' => function (EntityRepository $entityRepository) {
+                    return $entityRepository->createQueryBuilder('art')
```
Comment (thomasbisignani) [inline src/Form/ApplicationRequest/ApplicationRequestTagsType.php]: Not a big deal, but you can inline here :

```php
return $entityRepository->createQueryBuilder('art')->orderBy('art.name');
```

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 10]
```diff
@@ -110,4 +116,79 @@ public function municipalVolunteerDetailAction(
...
+        return $this->handleApplicationRequestTagsType(
+            $objectManager,
+            $request,
+            $volunteerRequest,
+            'municipal_chief/municipal/volunteer/edit_tags.html.twig',
+            'app_municipal_chief_municipal_volunteer_request'
+        );
+    }
+
+    private function handleApplicationRequestTagsType(
```
Comment (ottaviano) [inline src/Controller/EnMarche/MunicipalChiefController.php]: `handleApplicationRequestTagsType` => `handleApplicationRequestTagsRequest` wdyt?

**Pull request merged**